### PR TITLE
refactor components & create UploadModal

### PR DESCRIPTION
**Pull request by brendanashworth**
This is the first step of https://github.com/minio/miniobrowser/issues/179.

/cc @krishnasrinivas

### CODE REVIEW

File: app/js/components/ObjectsList.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react'
+import Moment from 'moment'
+import humanize from 'humanize'
+import connect from 'react-redux/lib/components/connect'
+
```
Comment (krishnasrinivas) [inline app/js/components/ObjectsList.js]: you could use function component style like how it was before:
https://medium.com/@housecor/react-stateless-functional-components-nine-wins-you-might-have-overlooked-997b0d933dbc#.v8eulfldy

Was there any reason you changed it?

**Comment by krishnasrinivas** [comment]
other than the one comment rest of the things look good.

File: app/js/components/ObjectsList.js (hunk 1 above)
Comment (brendanashworth) [inline app/js/components/ObjectsList.js]: Ah! I didn't realize that was intentional. I'll change the simple ones back.

**Comment by brendanashworth** [comment]
Fixed, ptal @krishnasrinivas

**Comment by brendanashworth** [comment]
Thanks!

**Comment by krishnasrinivas** [comment]
> Thanks!

no, thank you ;)

**Pull request merged**